src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -195,7 +193,7 @@ protected EC2Cloud(String id, boolean useInstanceProfileForCredentials, String c
 
     public abstract URL getS3EndpointUrl() throws IOException;
 
-    protected Object readResolve() {
+    private Object readResolve() {
```

Comment:
As this class is abstract, I think this should remain `protected`.

Reply:
Alright, reverted that part